### PR TITLE
Remove old build constraints

### PR DESCRIPTION
**Pull request by kleimkuhler**
#7371 upgraded the Go version which included using the new formats for [build constraints](https://pkg.go.dev/cmd/go#hdr-Build_constraints). This removes the old ones that are no longer used.

Signed-off-by: Kevin Leimkuhler <kleimkuhler@icloud.com>

### CODE REVIEW

**Review by olix0r — DISMISSED** [review]

**Comment by alpeb** [comment]
This means people need to switch to go 1.17 in their dev machines, right?

**Comment by olix0r** [comment]
Oh.. yeah... I don't have access to go1.17 yet, apparently

**Review by adleong — APPROVED** [review]
builds successfully for me on `go version go1.17.3 darwin/amd64`

**Comment by olix0r** [comment]
```
:; go version
go version go1.16.9 linux/amd64
:; go test ./cli/... -update
jaeger/cmd/install.go:11:2: //go:build comment without // +build comment
multicluster/cmd/allow.go:8:2: //go:build comment without // +build comment
pkg/charts/charts.go:10:2: //go:build comment without // +build comment
viz/cmd/install.go:15:2: //go:build comment without // +build comment
```

it's a bummer that this is backwards-incompatible. Since these lines aren't hurting anything, it may be better to wait to remove them to avoid the breakage.

**Comment by kleimkuhler** [comment]
I'll hold off until next week to merge this. If developers are still using 1.16, they _should_ be able to build the project, but there may be issues when trying to run certain `go mod` commands. Additionally, it's probably good to be using a matching Go version anyways now that the project uses 1.17.

**Review by alpeb — APPROVED** [review]

**Pull request merged**
